### ISSUE TITLE
Server error while creating Purchase Invoice from Purchase Receipt

### ISSUE THREAD

**Issue by pooja-pp8848** [8848digital/erpnext#651]
### Information about bug

![image](https://github.com/user-attachments/assets/b4f62550-ac09-4a8c-958c-227b4a78574b)

### Module

buying

### Version

ERPNext: v15.35.1 (pre-prod)

Frappe Framework: v15.41.0 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
### App Versions

{
	"assets": "0.0.1",
	"erpnext": "15.34.0",
	"frappe": "15.41.0",
	"payments": "0.0.1",
	"projects": "0.0.1",
	"sales_commission": "0.0.1"
}
```
### Route
```
Form/Purchase Invoice/ACC-PINV-2024-22314
```
### Traceback
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 114, in application
    response = frappe.api.handle(request)
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1775, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/desk/form/save.py", line 37, in savedocs
    doc.submit()
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1048, in submit
    return self._submit()
  File "apps/frappe/frappe/model/document.py", line 1031, in _submit
    return self.save()
  File "apps/frappe/frappe/model/document.py", line 337, in save
    return self._save(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 390, in _save
    self.run_post_save_methods()
  File "apps/frappe/frappe/model/document.py", line 1131, in run_post_save_methods
    self.run_method("on_submit")
  File "apps/frappe/frappe/model/document.py", line 962, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1322, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1304, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 959, in fn
    return method_object(*args, **kwargs)
  File "apps/erpnext/erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py", line 717, in on_submit
    self.make_gl_entries()
  File "apps/erpnext/erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py", line 754, in make_gl_entries
    make_gl_entries(
  File "apps/erpnext/erpnext/accounts/general_ledger.py", line 40, in make_gl_entries
    create_payment_ledger_entry(
  File "apps/erpnext/erpnext/accounts/utils.py", line 1776, in create_payment_ledger_entry
    ple.submit()
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1048, in submit
    return self._submit()
  File "apps/frappe/frappe/model/document.py", line 1031, in _submit
    return self.save()
  File "apps/frappe/frappe/model/document.py", line 337, in save
    return self._save(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 359, in _save
    return self.insert()
  File "apps/frappe/frappe/model/document.py", line 315, in insert
    self.run_post_save_methods()
  File "apps/frappe/frappe/model/document.py", line 1130, in run_post_save_methods
    self.run_method("on_update")
  File "apps/frappe/frappe/model/document.py", line 962, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1322, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1304, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 959, in fn
    return method_object(*args, **kwargs)
  File "apps/erpnext/erpnext/accounts/doctype/payment_ledger_entry/payment_ledger_entry.py", line 177, in on_update
    update_voucher_outstanding(
  File "apps/erpnext/erpnext/accounts/utils.py", line 1795, in update_voucher_outstanding
    voucher_outstanding = ple_query.get_voucher_outstandings(vouchers, common_filter=common_filter)
  File "apps/erpnext/erpnext/accounts/utils.py", line 2082, in get_voucher_outstandings
    self.query_for_outstanding()
  File "apps/erpnext/erpnext/accounts/utils.py", line 2045, in query_for_outstanding
    self.voucher_outstandings = self.cte_query_voucher_amount_and_outstanding.run(as_dict=True)
  File "apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
  File "apps/frappe/frappe/database/postgres/database.py", line 310, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "apps/frappe/frappe/database/database.py", line 227, in sql
    self._cursor.execute(query, values)
psycopg2.errors.GroupingError: column "tabPayment Ledger Entry.account" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: WITH vouchers AS (SELECT "account","voucher_type","voucher_n...
                                 ^


```
### Request Data
```
{
	"type": "POST",
	"args": {
		"doc": "{\"name\":\"ACC-PINV-2024-22310\",\"owner\":\"Administrator\",\"creation\":\"2024-11-18 15:38:17.697867\",\"modified\":\"2024-11-18 15:38:17.697867\",\"modified_by\":\"Administrator\",\"docstatus\":0,\"idx\":0,\"title\":\"_Test Supplier USD\",\"naming_series\":\"ACC-PINV-.YYYY.-\",\"supplier\":\"_Test Supplier USD\",\"supplier_name\":\"_Test Supplier USD\",\"tax_id\":null,\"company\":\"PP Ltd\",\"posting_date\":\"2024-11-18\",\"posting_time\":\"15:38:17.764225\",\"set_posting_time\":0,\"due_date\":\"2024-11-18\",\"is_paid\":0,\"is_return\":0,\"return_against\":null,\"update_outstanding_for_self\":1,\"update_billed_amount_in_purchase_order\":0,\"update_billed_amount_in_purchase_receipt\":1,\"apply_tds\":0,\"tax_withholding_category\":null,\"amended_from\":null,\"bill_no\":null,\"bill_date\":null,\"location\":null,\"cost_center\":null,\"department\":null,\"project\":null,\"currency\":\"USD\",\"conversion_rate\":62.9,\"use_transaction_date_exchange_rate\":1,\"buying_price_list\":\"Standard Buying\",\"price_list_currency\":\"INR\",\"plc_conversion_rate\":1,\"ignore_pricing_rule\":0,\"scan_barcode\":null,\"update_stock\":0,\"set_warehouse\":\"Finished Goods - PP Ltd\",\"set_from_warehouse\":null,\"is_subcontracted\":0,\"rejected_warehouse\":null,\"supplier_warehouse\":null,\"total_qty\":5,\"total_net_weight\":0,\"base_total\":1729.75,\"base_net_total\":1729.75,\"total\":27.5,\"net_total\":27.5,\"tax_withholding_net_total\":27.5,\"base_tax_withholding_net_total\":1729.75,\"tax_category\":\"\",\"taxes_and_charges\":null,\"shipping_rule\":null,\"incoterm\":null,\"named_place\":null,\"base_taxes_and_charges_added\":0,\"base_taxes_and_charges_deducted\":0,\"base_total_taxes_and_charges\":0,\"taxes_and_charges_added\":0,\"taxes_and_charges_deducted\":0,\"total_taxes_and_charges\":0,\"base_grand_total\":1729.75,\"base_rounding_adjustment\":0,\"base_rounded_total\":1729.75,\"base_in_words\":\"INR One Thousand, Seven Hundred And Twenty Nine and Seventy Five Paisa only.\",\"grand_total\":27.5,\"rounding_adjustment\":0,\"use_company_roundoff_cost_center\":0,\"rounded_total\":27.5,\"in_words\":\"USD Twenty Seven and Fifty Cent only.\",\"total_advance\":0,\"outstanding_amount\":27.5,\"disable_rounded_total\":0,\"apply_discount_on\":\"Grand Total\",\"base_discount_amount\":0,\"additional_discount_percentage\":0,\"discount_amount\":0,\"other_charges_calculation\":null,\"mode_of_payment\":null,\"base_paid_amount\":0,\"clearance_date\":null,\"cash_bank_account\":null,\"paid_amount\":0,\"allocate_advances_automatically\":0,\"only_include_allocated_payments\":0,\"write_off_amount\":0,\"base_write_off_amount\":0,\"write_off_account\":null,\"write_off_cost_center\":null,\"supplier_address\":null,\"address_display\":null,\"contact_person\":null,\"contact_display\":null,\"contact_mobile\":null,\"contact_email\":null,\"shipping_address\":\"PP-MH-Billing\",\"shipping_address_display\":\"001<br>\\nFort<br>Mumbai<br>\\nMaharashtra<br>400001<br>India<br>\\n<br>\\n\",\"billing_address\":\"PP-MH-Billing\",\"billing_address_display\":\"001<br>\\nFort<br>Mumbai<br>\\nMaharashtra<br>400001<br>India<br>\\n<br>\\n\",\"payment_term\":null,\"discount_due_date\":null,\"payment_terms_template\":null,\"ignore_default_payment_terms_template\":0,\"tc_name\":null,\"terms\":null,\"status\":\"Draft\",\"per_received\":100,\"credit_to\":\"Creditors - USD - PP Ltd\",\"party_account_currency\":\"USD\",\"is_opening\":\"No\",\"against_expense_account\":\"Stock Received But Not Billed - PP Ltd\",\"unrealized_profit_loss_account\":null,\"subscription\":null,\"auto_repeat\":null,\"from_date\":null,\"to_date\":null,\"letter_head\":\"PP Ltd\",\"group_same_items\":0,\"select_print_heading\":null,\"language\":\"en\",\"on_hold\":0,\"release_date\":null,\"hold_comment\":null,\"is_internal_supplier\":0,\"represents_company\":\"\",\"supplier_group\":\"_Test Supplier Group\",\"inter_company_invoice_reference\":null,\"is_old_subcontracting_flow\":0,\"remarks\":null,\"doctype\":\"Purchase Invoice\",\"tax_withheld_vouchers\":[],\"supplied_items\":[],\"pricing_rules\":[],\"advances\":[],\"payment_discount_terms\":[],\"items\":[{\"name\":\"td5hcb9v8o\",\"owner\":\"Administrator\",\"creation\":\"2024-11-18 15:38:17.697867\",\"modified\":\"2024-11-18 15:38:17.697867\",\"modified_by\":\"Administrator\",\"docstatus\":0,\"idx\":1,\"item_code\":\"PP Item 25\",\"product_bundle\":null,\"item_name\":\"PP Item 25\",\"description\":\"PP Item 25\",\"brand\":null,\"item_group\":\"8848\",\"image\":\"\",\"received_qty\":5,\"qty\":5,\"rejected_qty\":0,\"uom\":\"Nos\",\"conversion_factor\":1,\"stock_uom\":\"Nos\",\"stock_qty\":5,\"price_list_rate\":0,\"base_price_list_rate\":0,\"margin_type\":\"\",\"margin_rate_or_amount\":0,\"rate_with_margin\":0,\"discount_percentage\":0,\"discount_amount\":0,\"base_rate_with_margin\":0,\"rate\":5.5,\"amount\":27.5,\"item_tax_template\":null,\"base_rate\":345.95,\"base_amount\":1729.75,\"pricing_rules\":null,\"stock_uom_rate\":5.5,\"is_free_item\":0,\"apply_tds\":1,\"net_rate\":5.5,\"net_amount\":27.5,\"base_net_rate\":345.95,\"base_net_amount\":1729.75,\"valuation_rate\":345.95,\"sales_incoming_rate\":0,\"item_tax_amount\":0,\"landed_cost_voucher_amount\":0,\"rm_supp_cost\":0,\"warehouse\":\"Finished Goods - PP Ltd\",\"serial_and_batch_bundle\":null,\"use_serial_batch_fields\":1,\"from_warehouse\":null,\"quality_inspection\":null,\"rejected_warehouse\":null,\"rejected_serial_and_batch_bundle\":null,\"serial_no\":null,\"rejected_serial_no\":null,\"batch_no\":null,\"manufacturer\":null,\"manufacturer_part_no\":null,\"expense_account\":\"Stock Received But Not Billed - PP Ltd\",\"wip_composite_asset\":null,\"is_fixed_asset\":0,\"asset_location\":null,\"asset_category\":null,\"deferred_expense_account\":null,\"service_stop_date\":null,\"enable_deferred_expense\":0,\"service_start_date\":null,\"service_end_date\":null,\"allow_zero_valuation_rate\":0,\"item_tax_rate\":\"{}\",\"bom\":null,\"include_exploded_items\":0,\"purchase_invoice_item\":null,\"purchase_order\":null,\"po_detail\":null,\"purchase_receipt\":\"MAT-PRE-2024-07146\",\"pr_detail\":\"tcm7182dj4\",\"sales_invoice_item\":null,\"material_request\":null,\"material_request_item\":null,\"weight_per_unit\":0,\"total_weight\":0,\"weight_uom\":null,\"location\":null,\"project\":null,\"work_breakdown_structure\":null,\"cost_center\":\"Main - PP Ltd\",\"department\":null,\"project_name\":null,\"wbs_name\":null,\"page_break\":0,\"parent\":\"ACC-PINV-2024-22310\",\"parentfield\":\"items\",\"parenttype\":\"Purchase Invoice\",\"doctype\":\"Purchase Invoice Item\",\"__unsaved\":1}],\"taxes\":[],\"advance_tax\":[],\"payment_schedule\":[{\"name\":\"td5kvs4cj6\",\"owner\":\"Administrator\",\"creation\":\"2024-11-18 15:38:18.050764\",\"modified\":\"2024-11-18 15:38:18.050764\",\"modified_by\":\"Administrator\",\"docstatus\":0,\"idx\":1,\"payment_term\":null,\"description\":null,\"due_date\":\"2024-11-18\",\"mode_of_payment\":null,\"invoice_portion\":100,\"discount_type\":null,\"discount_date\":null,\"discount\":0,\"payment_amount\":27.5,\"outstanding\":27.5,\"paid_amount\":0,\"discounted_amount\":0,\"base_payment_amount\":1729.75,\"parent\":\"ACC-PINV-2024-22310\",\"parentfield\":\"payment_schedule\",\"parenttype\":\"Purchase Invoice\",\"doctype\":\"Payment Schedule\"}],\"__onload\":{\"make_payment_via_journal_entry\":0,\"backflush_based_on\":\"BOM\",\"supplier_tds\":null},\"__last_sync_on\":\"2024-11-18T10:08:17.188Z\"}",
		"action": "Submit"
	},
	"btn": {
		"jQuery3700138496374871359681": {
			"events": {
				"click": [
					{
						"type": "click",
						"origType": "click",
						"guid": 53736,
						"namespace": ""
					}
				]
			}
		}
	},
	"freeze": true,
	"headers": {},
	"error_handlers": {},
	"url": "/api/method/frappe.desk.form.save.savedocs",
	"request_id": null
}
```
### Response Data
```
{
	"exception": "",
	"exc_type": "GroupingError",
	"_exc_source": "erpnext (app)",
	"_debug_messages": "[\"Error in query:\\ncolumn \\\"tabPayment Ledger Entry.account\\\" must appear in the GROUP BY clause or be used in an aggregate function\\nLINE 1: WITH vouchers AS (SELECT \\\"account\\\",\\\"voucher_type\\\",\\\"voucher_n...\\n                                 ^\\n\"]"
}
```
```

**Comment by anandk-spyke-o1** [comment]
FIxed in #653

**Comment by tinadn** [comment]
Resolved. Hence closing.
